attributioncode/validator.go

```diff
@@ -38,6 +53,12 @@ func NewValidator(hmacKey string, timeout time.Duration) *Validator {
 // Validate validates and sanitizes attribution code and signature
 func (v *Validator) Validate(code, sig string) (string, error) {
 	logEntry := logrus.WithField("b64code", code)
+
+	if code == "" {
+		logEntry.Error("code is empty")
+		return "", errors.New("code is empty")
```

Comment:
When does this trigger? Or is the `validCodes` test with `""` wrong? Should empty code result in the 3 "not set" and "other" source?

Reply:
Oh I see fixed in https://github.com/mozilla-services/stubattribution/pull/95/commits/5cfc62fe735b422ea1b04be3bdcc1d190b84256d

Reply:
This only triggers if `attributioncode=""` or no attribution code parameter is in the query string. In that case, it just redirects to a standard non-attributed installer.

Reply:
I've also updated the tests.